packages/nextjs/app/batches/_components/UpdateBatchModalContent.tsx

```diff
@@ -0,0 +1,191 @@
+import { forwardRef, useMemo, useState } from "react";
+import { InputBase } from "~~/components/scaffold-eth";
+import { BatchStatus } from "~~/services/database/config/types";
+import { notification } from "~~/utils/scaffold-eth";
+
+export type BatchOperation = "add" | "edit";
+
+type UpdateBatchModalContentProps = {
+  modalId: string;
+  defaultName?: string;
+  defaultStatus?: BatchStatus;
+  defaultStartDate?: Date;
+  defaultTelegramLink?: string;
+  defaultRegistryAddress?: string;
+  isPending: boolean;
+  batchOperation: BatchOperation;
+  updateBatch: (data: {
+    name: string;
+    status: BatchStatus;
+    startDate: string;
+    telegramLink: string;
+    contractAddress: string;
+    bgSubdomain: string;
+  }) => void;
+};
+
+export const UpdateBatchModalContent = forwardRef<HTMLInputElement, UpdateBatchModalContentProps>(
```

Comment:
nit: Not sure update is the right word? Lol but also can't come with better name. 

Since this modal is used for both create and update batch

Reply:
Ahh I see, we have this to display Conents like "UI" only component. 

Shall we call this just "BatchModalContent"? Because its UI only and updattion / creation functions are passed to it? 

And maybe rename EditModal to UpdateBatchModal? And also inline stuff with https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/137#discussion_r2054303239?